src/test/java/com/hpe/cloudfoundryjenkins/CloudFoundryPushPublisherTest.java

```diff
@@ -150,18 +150,17 @@ public void testPerformSimplePushJenkinsConfig() throws Exception {
 
     @Test
     @WithTimeout(300)
-    public void testPerformResetIfExists() throws Exception {
+public void testPerformCutoverMethod() throws Exception {
```

Comment:
It would be better if each cutover method was tested individually, as in, testPerformCutoverDeploy(), testPerformCutoverRedeploy(), and testPerformCutoverReroute().

You should also test if each cutover method worked correctly, checking if the build succeeded is just the baseline. 
testPerformCutoverDeploy() should deploy an app, then try deploying it again and test that the second app was NOT deployed.
testPerformCutoverRedeploy() should be basically the same thing as the old testPerformResetIfExists() was.
testPerformCutoverReroute() is a bit trickier to test since the end result is the same as "redeploy". Maybe something similar to redeploy would be good enough.